sdk/src/impl/position-impl.ts

```diff
@@ -315,17 +362,57 @@ export class PositionImpl implements Position {
     );
 
     const accountExemption = await this.ctx.fetcher.getAccountRentExempt();
-    const { ataTokenAddresses, resolveAtaIxs } = await resolveAtaForMints(this.ctx, {
-      mints: initializedRewards.map((r) => r.mint),
-      accountExemption,
-      receiver: destinationWalletKey,
-      payer: ataPayerKey,
-    });
+    const affiliatedMints = getTokenMintsFromWhirlpools([whirlpool]);
+    const walletTokenAccountsByMint = { ...ownerTokenAccountsRecord };
+
+    const solInRewardMints = initializedRewards.some((info) => TokenUtil.isNativeMint(info.mint));
+    const solInWalletTokenAccounts = !!walletTokenAccountsByMint[NATIVE_MINT.toBase58()];
+    const mintsToResolveAtasFor = affiliatedMints.filter(
+      // Filter out mints for which we already have token accounts
+      (mint) => !walletTokenAccountsByMint[mint.toBase58()]
+    );
```

Comment:
can we modify `getTokenMintsFromWhirlpools` to accept a flag that can return you `all-tokens`, `pool-tokens` & `reward-tokens` only?

That way we can reduce duplicated code here.

ResolveAta also automatically handles mints that already has an ATA, so you don't have to perform the check here.